prime-router/src/main/kotlin/azure/ReportFunction.kt

```diff
@@ -63,24 +63,33 @@ class ReportFunction {
     ): HttpResponseMessage {
         try {
             val workflowEngine = WorkflowEngine()
+            var actionHistory = ActionHistory("receive", context)
+            actionHistory.trackActionParams(request)
             val validatedRequest = validateRequest(workflowEngine, request)
-            when {
+            val httpResponseMessage = when {
                 validatedRequest.options == Options.CheckConnections -> {
                     workflowEngine.checkConnections()
-                    return okResponse(request, validatedRequest)
+                    okResponse(request, validatedRequest)
                 }
                 validatedRequest.report == null -> {
-                    return badRequestResponse(request, validatedRequest)
+                    badRequestResponse(request, validatedRequest)
                 }
                 validatedRequest.options == Options.ValidatePayload -> {
-                    return okResponse(request, validatedRequest)
+                    okResponse(request, validatedRequest)
+                }
+                else -> {
+                    context.logger.info("Successfully reported: ${validatedRequest.report!!.id}.")
+                    val destinations = mutableListOf<String>()
+                    routeReport(context, workflowEngine, validatedRequest, destinations, actionHistory)
+                    val responseBody = createResponseBody(validatedRequest, destinations)
+                    workflowEngine.receiveReport(validatedRequest.report)
+                    actionHistory.trackExternalIncomingReport(validatedRequest)
+                    createdResponse(request, validatedRequest, responseBody)
                 }
             }
-            context.logger.info("Successfully reported: ${validatedRequest.report!!.id}.")
-            workflowEngine.receiveReport(validatedRequest.report)
-            val destinations = mutableListOf<String>()
-            routeReport(context, workflowEngine, validatedRequest, destinations)
-            return createdResponse(request, validatedRequest, destinations)
+            actionHistory.trackActionResult(httpResponseMessage)
```

Comment:
The above messy code reshuffling  is to set up for this call to trackActionResults - now tracking not just for submissions, but now _every_ call to reports/api endpoint that passes authentication will get logged in the database.